crates/index-scheduler/src/lib.rs

```diff
@@ -366,6 +369,7 @@ impl IndexScheduler {
                     match ret {
                         Ok(Ok(TickOutcome::TickAgain(_))) => (),
                         Ok(Ok(TickOutcome::WaitForSignal)) => run.scheduler.wake_up.wait(),
+                        Ok(Ok(TickOutcome::StopProcessingForever)) => break,
```

Comment:
can we check that `run` is never called again?

Reply:
I checked manually and that seems to be the case.
If you were thinking about something automatic, I think we should open an issue and do it in another PR since the index-scheduler was already not supposed to work with two run functions executing at the same time.